### ISSUE TITLE
The performance of this code

### ISSUE THREAD

**Issue by qilong-zhang** [psandovalsegura/pytorch-gd-uap#1]
Hi, 

I implement this code and test on VGG-16, the result is about 85%. But the result from the original paper is only 63%. Could you help me? Why is this happening?

**Comment by psandovalsegura** [comment]
If you take a look at [the tensorflow code](https://github.com/val-iisc/GD-UAP), the authors provide [weights](https://github.com/val-iisc/GD-UAP/tree/master/classification/weights) for their VGG-16. I tried loading these weights in this implementation, but they negatively affected the VGG-16's accuracy on ImageNet. 

So, to answer your question, I think the reason the UAPs from this repo are more effective is due to the pre-trained weights  from pytorch. Let me know if that helps or if you find a different reason.

**Comment by qilong-zhang** [comment]
@psandovalsegura  I have a question about the `gduap.py`, In line 
```python
 loss = -sum(list(map(lambda activation: torch.log(torch.sum(torch.square(activation)) / 2), activations)))
```
why `torch.sum(torch.square(activation))` is divided by 2?

**Comment by psandovalsegura** [comment]
@qilong-zhang When I was writing this, I wanted it to be as similar as possible to the tensorflow code which uses `tf.nn.l2_loss` when [calculating the loss](https://github.com/val-iisc/GD-UAP/blob/master/classification/utils/losses.py#L31). The [documentation](https://www.tensorflow.org/api_docs/python/tf/nn/l2_loss) for this function says it "computes half the L2 norm of a tensor without the sqrt".

I agree that if you were to follow the paper exactly (Line 7 of Algorithm 1), it wouldn't be divided by 2. But it shouldn't make much of a difference.

**Comment by qilong-zhang** [comment]
@psandovalsegura Thanks for your code!
